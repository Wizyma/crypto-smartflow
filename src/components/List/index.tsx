import { Row, Col, Text, Divider, Card, Image, Capacity } from '@zeit-ui/react';
import React from 'react';

import useTotalAmount from 'hooks/useTotalAmount';
import { ICON_URL, CRYPTO_DICTIONNARY } from '../../constatns';
import { CryptoRsponse } from 'types/dataContract';

export default function List({
  cryptos,
  wallet,
}: {
  cryptos: CryptoRsponse;
  wallet: { [key: string]: number };
}) {
  const { total, percentage } = useTotalAmount({ cryptos, wallet });

  return (
    <>
      {percentage.map((item) => {
        const key = Object.keys(item)[0];

        return (
          <React.Fragment key={key}>
            <Card.Content>
              <Row gap={0.8}>
                <Col span={6}>
                  <Image src={ICON_URL(key)} width={50} height={50} alt={key} />
                </Col>
                <Col>
                  <Text>{CRYPTO_DICTIONNARY[key]}</Text>
                </Col>
                <Col
                  style={{
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'flex-end',
                    marginRight: 5,
                  }}
                >
                  <Capacity value={parseInt(item[key], 10)} />
                  <Text style={{ marginLeft: 15 }}>{item[key]} %</Text>
                </Col>
                <Col style={{ display: 'flex', justifyContent: 'flex-end' }}>
                  <Text>
                    {wallet[key]} {key}
                  </Text>
                </Col>
                <Col style={{ display: 'flex', justifyContent: 'flex-end' }}>
                  <Text style={{ color: '#ccc' }}>{(cryptos[key].EUR * wallet[key]).toFixed(2)} €</Text>
                </Col>
              </Row>
            </Card.Content>
            <Divider y={0} />
          </React.Fragment>
        );
      })}
      <Card.Content>
        <Text style={{ color: '#ccc', textAlign: 'center', fontWeight: 700 }}>Total Amount = {total} €</Text>
      </Card.Content>
    </>
  );
}
