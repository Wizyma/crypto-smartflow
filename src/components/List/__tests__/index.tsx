import { render } from '@testing-library/react';
import React from 'react';

import List from 'components/List';
import data from '../../../__fixtures__';

test('render List component correctly', () => {
  const { asFragment } = render(<List {...data} />);

  expect(asFragment()).toMatchSnapshot();
});
