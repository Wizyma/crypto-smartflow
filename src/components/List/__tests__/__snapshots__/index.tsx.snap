// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render List component correctly 1`] = `
<DocumentFragment>
  <div
    class="jsx-2956695818 content "
  >
    <div
      class="jsx-1376450318 row "
    >
      <div
        class="jsx-158700623 col "
      >
        <div
          class="jsx-121907836 image "
        >
          <div
            class="jsx-2292771821 skeleton"
          />
          <img
            alt="BTC"
            class="jsx-121907836 "
            height="50"
            src="https://cryptoicons.org/api/color/btc/200"
            width="50"
          />
        </div>
      </div>
      <div
        class="jsx-180329529 col "
      >
        <p
          class="jsx-3390086683  "
        >
          Bitcoin
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; align-items: center; justify-content: flex-end; margin-right: 5px;"
      >
        <div
          class="jsx-898285201 capacity "
          title="29%"
        >
          <span
            class="jsx-898285201"
          />
        </div>
        <p
          class="jsx-3390086683  "
          style="margin-left: 15px;"
        >
          29.26 %
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
        >
          0.1005 BTC
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
          style="color: rgb(204, 204, 204);"
        >
          860.98 €
        </p>
      </div>
    </div>
  </div>
  <div
    class="jsx-3382564211 divider "
    role="separator"
  />
  <div
    class="jsx-2956695818 content "
  >
    <div
      class="jsx-1376450318 row "
    >
      <div
        class="jsx-158700623 col "
      >
        <div
          class="jsx-121907836 image "
        >
          <div
            class="jsx-2292771821 skeleton"
          />
          <img
            alt="ETH"
            class="jsx-121907836 "
            height="50"
            src="https://cryptoicons.org/api/color/eth/200"
            width="50"
          />
        </div>
      </div>
      <div
        class="jsx-180329529 col "
      >
        <p
          class="jsx-3390086683  "
        >
          Etherum
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; align-items: center; justify-content: flex-end; margin-right: 5px;"
      >
        <div
          class="jsx-1520919870 capacity "
          title="15%"
        >
          <span
            class="jsx-1520919870"
          />
        </div>
        <p
          class="jsx-3390086683  "
          style="margin-left: 15px;"
        >
          15.19 %
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
        >
          2.2349 ETH
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
          style="color: rgb(204, 204, 204);"
        >
          446.98 €
        </p>
      </div>
    </div>
  </div>
  <div
    class="jsx-3382564211 divider "
    role="separator"
  />
  <div
    class="jsx-2956695818 content "
  >
    <div
      class="jsx-1376450318 row "
    >
      <div
        class="jsx-158700623 col "
      >
        <div
          class="jsx-121907836 image "
        >
          <div
            class="jsx-2292771821 skeleton"
          />
          <img
            alt="DASH"
            class="jsx-121907836 "
            height="50"
            src="https://cryptoicons.org/api/color/dash/200"
            width="50"
          />
        </div>
      </div>
      <div
        class="jsx-180329529 col "
      >
        <p
          class="jsx-3390086683  "
        >
          DASH
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; align-items: center; justify-content: flex-end; margin-right: 5px;"
      >
        <div
          class="jsx-2131296490 capacity "
          title="0%"
        >
          <span
            class="jsx-2131296490"
          />
        </div>
        <p
          class="jsx-3390086683  "
          style="margin-left: 15px;"
        >
          0.01 %
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
        >
          0.2527 DASH
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
          style="color: rgb(204, 204, 204);"
        >
          0.17 €
        </p>
      </div>
    </div>
  </div>
  <div
    class="jsx-3382564211 divider "
    role="separator"
  />
  <div
    class="jsx-2956695818 content "
  >
    <div
      class="jsx-1376450318 row "
    >
      <div
        class="jsx-158700623 col "
      >
        <div
          class="jsx-121907836 image "
        >
          <div
            class="jsx-2292771821 skeleton"
          />
          <img
            alt="BAT"
            class="jsx-121907836 "
            height="50"
            src="https://cryptoicons.org/api/color/bat/200"
            width="50"
          />
        </div>
      </div>
      <div
        class="jsx-180329529 col "
      >
        <p
          class="jsx-3390086683  "
        >
          Basic Attention Token
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; align-items: center; justify-content: flex-end; margin-right: 5px;"
      >
        <div
          class="jsx-2889423309 capacity "
          title="7%"
        >
          <span
            class="jsx-2889423309"
          />
        </div>
        <p
          class="jsx-3390086683  "
          style="margin-left: 15px;"
        >
          7.21 %
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
        >
          82.4851 BAT
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
          style="color: rgb(204, 204, 204);"
        >
          211.99 €
        </p>
      </div>
    </div>
  </div>
  <div
    class="jsx-3382564211 divider "
    role="separator"
  />
  <div
    class="jsx-2956695818 content "
  >
    <div
      class="jsx-1376450318 row "
    >
      <div
        class="jsx-158700623 col "
      >
        <div
          class="jsx-121907836 image "
        >
          <div
            class="jsx-2292771821 skeleton"
          />
          <img
            alt="USDC"
            class="jsx-121907836 "
            height="50"
            src="https://cryptoicons.org/api/color/usdc/200"
            width="50"
          />
        </div>
      </div>
      <div
        class="jsx-180329529 col "
      >
        <p
          class="jsx-3390086683  "
        >
          USD Coin
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; align-items: center; justify-content: flex-end; margin-right: 5px;"
      >
        <div
          class="jsx-4083258065 capacity "
          title="48%"
        >
          <span
            class="jsx-4083258065"
          />
        </div>
        <p
          class="jsx-3390086683  "
          style="margin-left: 15px;"
        >
          48.33 %
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
        >
          14.5099 USDC
        </p>
      </div>
      <div
        class="jsx-180329529 col "
        style="display: flex; justify-content: flex-end;"
      >
        <p
          class="jsx-3390086683  "
          style="color: rgb(204, 204, 204);"
        >
          1421.97 €
        </p>
      </div>
    </div>
  </div>
  <div
    class="jsx-3382564211 divider "
    role="separator"
  />
  <div
    class="jsx-2956695818 content "
  >
    <p
      class="jsx-3390086683  "
      style="color: rgb(204, 204, 204); text-align: center; font-weight: 700;"
    >
      Total Amount = 2942.09 €
    </p>
  </div>
</DocumentFragment>
`;
